model/patch/trigger_intent.go

```diff
@@ -0,0 +1,104 @@
+package patch
+
+import (
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	mgobson "gopkg.in/mgo.v2/bson"
+)
+
+const TriggerIntentType = "trigger"
+
+type TriggerIntent struct {
+	Id             string                   `bson:"_id"`
+	Requester      string                   `bson:"requester"`
+	Author         string                   `bson:"author"`
+	ProjectID      string                   `bson:"project_id"`
+	ParentID       string                   `bson:"parent_id"`
+	ParentAsModule string                   `bson:"parent_as_module"`
+	ParentStatus   string                   `bson:"parent_status"`
+	Definitions    []PatchTriggerDefinition `bson:"definitions"`
+
+	Processed bool `bson:"processed"`
+}
+
+var (
+	triggerIDKey        = bsonutil.MustHaveTag(TriggerIntent{}, "Id")
+	triggerProcessedKey = bsonutil.MustHaveTag(TriggerIntent{}, "Processed")
+)
+
+func (t *TriggerIntent) ID() string {
+	return t.Id
+}
+
+func (t *TriggerIntent) Insert() error {
+	return errors.Wrap(db.Insert(IntentCollection, t), "problem inserting trigger intent")
+}
+
+func (t *TriggerIntent) SetProcessed() error {
+	t.Processed = true
+	return updateOneIntent(
+		bson.M{triggerIDKey: t.Id},
+		bson.M{"$set": bson.M{
+			triggerProcessedKey: true,
+		}},
+	)
+}
+
+func (t *TriggerIntent) IsProcessed() bool {
+	return t.Processed
+}
+
+// GetType returns the patch intent, e.g., GithubIntentType.
+func (t *TriggerIntent) GetType() string {
+	return TriggerIntentType
+}
+
+func (t *TriggerIntent) NewPatch() *Patch {
+	return &Patch{
+		Id:       mgobson.ObjectIdHex(t.Id),
```

Comment:
Annie and I [had a discussion](https://github.com/evergreen-ci/evergreen/pull/4233#discussion_r536199161) about child patches' description. My thinking was that since they will be hidden to users it didn't make sense to give them a description.